src/engraving/dom/masterscore.cpp

```diff
@@ -656,7 +656,11 @@ MeasureBase* MasterScore::insertMeasure(MeasureBase* beforeMeasure, const Insert
                         if (ee) {
                             doUndoRemoveElement(ee);
                             if (s->empty()) {
-                                undoRemoveElement(s);
+                                if (ee->isTimeSig()) {
```

Comment:
I believe here you should add ` || ee->isKeySig()`, at least in Mu3 it otherwise fails at a unit test (which may not exist or be disabled in Mu4?)